### ISSUE TITLE
wireplumber 0.4.11-1 have no valid source/sink

### ISSUE THREAD

**Issue by pffang** [pipewire-debian/pipewire-debian#71]
**If you Completed all troubleshoot options, check below**     
- [√]  Did you Follow all installation instructions carefully from [here](../wiki/#1-ppa-configuration)?
- [√]  Did you check All troubleshoot options from [here](../wiki/Troubleshooting)?

**Desktop (please complete the following information):**
 - Distro: Ubuntu
 - Desktop: Gnome
 - Version: 20.04
 - PipeWire Related packages Version: wireplumber 0.4.11-1~ubuntu20.04
 - Kernel version: 5.13.0-52

**Describe the bug**
wireplumber 0.4.11-1 have no valid source/sink

**To Reproduce**
Just upgrade from an older version to version 0.4.11-1, pipewire lost the source/sink, only have a dummy output and input

**Expected behavior**
pipewire have the correct source/sink.

**Screenshots**
(snapshot of PW related services or anything)

**Journal log and status of related services**

**Additonal Information (If applicable):**                                          
Uninstall wireplumber 0.4.11-1 and install pipewire-media-session can fix this issue.

**Comment by souravdas142** [comment]
Do you have wireplumber/pipewire config files at your /home/.config or /etc ? can you comfirm this please ?

**Comment by JCallicoat** [comment]
I have the same issue. This is caused by this error on wireplumber start up:

Jul 06 20:42:15 somehost wireplumber[12700]: [string "alsa.lua"]:169: attempt to call a string value (local 'type')
                                                    stack traceback:
                                                            [string "alsa.lua"]:169: in function <[string "alsa.lua"]:52>


This appears to come from the call to `Core.get_vm_type()` not being defined here in /usr/share/wireplumber/scripts/monitors/alsa.lua:

```
  if nonempty(Core.get_vm_type()) and type(vm_overrides) == "table" then
    for k, v in pairs(vm_overrides) do
      properties[k] = v
    end
  end
```

It seems like the updated lua scripts were taken but this commit that adds that function was not:

https://gitlab.freedesktop.org/pipewire/wireplumber/-/commit/efaaf3b43a9d29c69fbe5a20b747eb365370d7d6

If I change that block to remove the `if` and only leave the `for` loop the error goes away and I have all my sinks and sources again.

```
  --if nonempty(Core.get_vm_type()) and type(vm_overrides) == "table" then
    for k, v in pairs(vm_overrides) do
      properties[k] = v
    end
  --end
```

**Comment by JCallicoat** [comment]
Actually, it looks like the real problem was shadowing lua's `type` builtin in the function declarations:

https://gitlab.freedesktop.org/pipewire/wireplumber/-/commit/c16e637c329bc9dda8544b18f5bd47a8d63ee253

I reverted my changes to /usr/share/wireplumber/scripts/monitors/alsa.lua and applied that and everything still works.

**Comment by pffang** [comment]
> 

Yes, I have.
I removed all the config files in ~/.config and then copy from /usr/share for the latest version, but this issue is still observed.

**Comment by souravdas142** [comment]
https://github.com/pipewire-debian/pipewire-debian/issues/71#issuecomment-1178246586

Thank you @JCallicoat . Is this solution working for all ?

**Comment by souravdas142** [comment]
@pffang  No need to copy any configs under /etc or ~/.config unless you have some valid reason.

**Comment by pffang** [comment]
@souravdas142 @JCallicoat WOW, this solution works for my Ubuntu 20.04 VM. and also fix the same issue in another ArchLinux VM.

So this is a bug in VM checking in upstream?

**Comment by pffang** [comment]
This issue is fixed in 0.4.11.r4.gaee4fe2-1~ubuntu20.04
